Kitodo/src/main/java/org/kitodo/production/helper/tasks/HierarchyMigrationTask.java

```diff
@@ -0,0 +1,358 @@
+     */
+    private List<Integer> createParentProcess(Process childProcess)
+            throws ProcessGenerationException, IOException, DataException {
+
+        processGenerator.generateProcess(childProcess.getTemplate().getId(), childProcess.getProject().getId());
+        Process parentProcess = processGenerator.getGeneratedProcess();
+        processService.save(parentProcess);
+        fileService.createProcessLocation(parentProcess);
+        createParentMetsFile(childProcess);
+        ArrayList<Integer> parentData = new ArrayList<>();
+        parentData.add(parentProcess.getId());
+        URI metadataFilePath = fileService.getMetadataFilePath(childProcess);
+        parentData.add(convertChildMetsFile(metadataFilePath));
+        linkInDatabase(parentProcess, childProcess);
+        return parentData;
+    }
+
+    /**
+     * Links parent process and child process in the database. The processes are
+     * saved.
+     *
+     * @param parentProcess
+     *            parent process to link
+     * @param childProcess
+     *            child process to link
+     */
+    private static void linkInDatabase(Process parentProcess, Process childProcess) throws DataException {
+        parentProcess.getChildren().add(childProcess);
+        childProcess.setParent(parentProcess);
+        processService.save(childProcess);
+    }
+
+    /**
+     * Generates the METS file for the parent process from the process anchor
+     * file.
+     *
+     * @param process
+     *            process to migrate
+     */
+    private void createParentMetsFile(Process process) throws IOException {
+        URI metadataFileUri = fileService.getMetadataFilePath(process);
+        URI anchorFileUri = fileService.createAnchorFile(metadataFileUri);
+        Workpiece workpiece = metsService.loadWorkpiece(anchorFileUri);
+        LinkedMetsResource link = workpiece.getRootElement().getChildren().get(0).getLink();
+        link.setLoctype("Kitodo.Production");
+        link.setUri(processService.getProcessURI(process));
+        URI parentMetadataFileUri = fileService.getMetadataFilePath(processGenerator.getGeneratedProcess(), false,
+            false);
+        metsService.saveWorkpiece(workpiece, parentMetadataFileUri);
+    }
+
+    /**
+     * Changes the METS file of the child process.
+     *
+     * @param metadataFilePath
+     *            URI of the metadata file
+     * @return the current number, may be {@code null}
+     */
+    private static Integer convertChildMetsFile(URI metadataFilePath) throws IOException {
+        Workpiece workpiece = metsService.loadWorkpiece(metadataFilePath);
+        IncludedStructuralElement childStructureRoot = workpiece.getRootElement().getChildren().get(0);
+        workpiece.setRootElement(childStructureRoot);
+        metsService.saveWorkpiece(workpiece, metadataFilePath);
+        return getCurrentNo(childStructureRoot);
+    }
+
+    /**
+     * Extracts the CurrentNo from the metadata.
```

Comment:
What does currentNO stands for in the metadata?
Please give a more explicit description

Reply:
The current number is an integer sorting criterion, which specifies the order of the subordinate units within their superordinate entirety. In case of journal issues, this can be the same as the issue number if the issue number continues to be counted at the turn of the year. In the case of multi-volume works, this number can correspond to the part number, or (for lexica, for example) it is counted up according to the alphabetical order of the volumes, supplementary volumes are counted on afterwards (thus, in the order in which the books are usually placed on a shelf).